From 1c5b33a4f92964ee0fd56dbd06351d0b83a27962 Mon Sep 17 00:00:00 2001
From: Paul Moore <pmoore@redhat.com>
Date: Thu, 4 Sep 2014 14:49:03 -0400
Subject: [security] selinux: fix a problem with IPv6 traffic denials in selinux_ip_postroute()

Message-id: <20140904144903.26034.19198.stgit@localhost>
Patchwork-id: 91232
O-Subject: [RHEL7 PATCH] selinux: fix a problem with IPv6 traffic denials in selinux_ip_postroute()
Bugzilla: 1102641
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Eric Paris <eparis@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1102641
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7914248
Upstream: Present in the SELinux tree, see the commit ID below
Testing: Tested, see notes in BZ

 commit a7a91a1928fe69cc98814cb746d5171ae14d757e
 Author: Paul Moore <pmoore@redhat.com>
 Date:   Wed Sep 3 10:51:59 2014 -0400

 selinux: fix a problem with IPv6 traffic denials in selinux_ip_postroute()

 A previous commit c0828e50485932b7e019df377a6b0a8d1ebd3080 ("selinux:
 process labeled IPsec TCP SYN-ACK packets properly in
 selinux_ip_postroute()") mistakenly left out a 'break' from a switch
 statement which caused problems with IPv6 traffic.

 Thanks to Florian Westphal for reporting and debugging the issue.

 Reported-by: Florian Westphal <fwestpha@redhat.com>
 Signed-off-by: Paul Moore <pmoore@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
index fd7cc9b..14d3fce 100644
--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -4903,6 +4903,7 @@ static unsigned int selinux_ip_postroute(struct sk_buff *skb, int ifindex,
    case PF_INET6:
     if (IP6CB(skb)->flags & IP6SKB_XFRM_TRANSFORMED)
      return NF_ACCEPT;
+    break;
    default:
     return NF_DROP_ERR(-ECONNREFUSED);
    }
-- 
1.7.1